src/monitorAnimationSpeed.js

```diff
@@ -0,0 +1,43 @@
+import React, {Component} from 'react'
+import collectFps from 'collect-fps'
+
+const noop = () => {}
+
+export default ({
+  threshold = 30,
+  fpsCollector = collectFps
+}) => (Target) => {
+  let endFPSCollection = noop
+
+  const handleStartFPSCollection = (component) => {
+    if (endFPSCollection !== noop) { endFPSCollection() }
+
+    endFPSCollection = fpsCollector()
+  }
+
+  class MonitorAnimationSpeed extends Component {
+    constructor () {
+      super()
+
+      this.state = {fps: 60}
+    }
+
+    render () {
+      return <Target
+        {...this.props}
+        onStartFPSCollection={handleStartFPSCollection}
+        onEndFPSCollection={() => {
+          if (endFPSCollection !== noop) {
+            this.setState({fps: endFPSCollection()})
+            endFPSCollection = noop
+          }
+        }}
+        lowFPS={this.state.fps < threshold}
+      />
+    }
+  }
+
+  MonitorAnimationSpeed.displayName = Target.displayName || Target.name
```

Comment:
About the

```
|- Slider 
|  |-- Slider
```

situation, this is the same for all the higher-order components in this library. I think the point is that we want to add the features that the components provide "seamlessly". I don’t know if this is the right approach. I haven’t optimized for the React dev tools because I don’t use them myself so the only thing that becomes visible to me is the output of [react-element-to-jsx-string](https://www.npmjs.com/package/react-element-to-jsx-string)

Maybe we should review this approach project wide and set the name manually on our components, after passing them through the higher-order components composition.